client/my-sites/domains/domain-management/settings/index.tsx

```diff
@@ -110,6 +110,12 @@ const Settings = ( {
 		if ( ! isSecuredWithUs( domain ) ) {
 			return null;
 		}
+		if (
+			domain.type === domainTypes.SITE_REDIRECT ||
```

Comment:
```suggestion
			domain.type === domainTypes.SITE_REDIRECT ||
			domain.type === domainTypes.TRANSFER ||
```

I think we can also hide this section if `domain.type` is a transfer:

<img width="718" alt="Screenshot 2023-07-07 at 17 58 02" src="https://github.com/Automattic/wp-calypso/assets/5324818/adda210d-1f53-4d00-8b07-9215c11db42c">